docs/pages/machine-id/reference/github-actions.mdx

```diff
@@ -0,0 +1,105 @@
+cluster. Joining clients must specify which `token` they want to use, and then
+information included in their join request is compared to the rules contained
+within the token by the Auth Server to determine whether or not they should be
+admitted.
+
+The following snippet shows all available options in the `token` resource when
+used for GitHub joining with Machine ID:
+
+```yaml
+kind: token
+version: v2
+metadata:
+  # name identifies the token. When configuring a bot or node to join using this
+  # token, this name should be specified.
+  name: github-token
+  # expires is a time in the future that this token should no longer be
+  # accepted. For GitHub joining, there is no need to set this to a short value.
+  expires: "3000-01-01T00:00:00Z"
+spec:
+  # For Machine ID and GitHub joining, roles will always be `Bot` and
+  # join_method will always be `github`.
+  roles: [Bot]
+  join_method: github
+
+  # bot_name specifies the name of the bot that this token will grant access to
+  # when it is used.
+  bot_name: github-demo
+
+  github:
+    # enterprise_server_host allows joining from GitHub Actions workflows in a
+    # GitHub Enterprise Server instance. For normal situations, where you are
+    # using github.com, this option should be omitted. If you are using GHES,
+    # this value should be configured to the hostname of your GHES instance.
+    enterprise_server_host: ghes.example.com
+
+    # allow is an array of rule configurations for what GitHub Actions workflows
+    # should be allowed to join. All options configured within one allow entry
+    # must be satisfied for the GitHub Actions run to be allowed to join. Where
+    # multiple allow entries are specified, any run which satisfies all of the
+    # options within a single entry will be allowed to join.
+    #
+    # An allow entry must include at least one of:
+    # - repository
+    # - repository_owner
+    # - sub
+    allow:
+      - # repository is full qualified (e.g including the owner) name of a
+        # GitHub repository.
+        repository: gravitational/teleport
+        # repository_owner is the name of an organization or user that a
+        # repository belongs to.
+        repository_owner: gravitational
+        # workflow is the exact name of a workflow as configured in the GitHub 
+        # Action workflow YAML file.
+        workflow: my-workflow
+        # environment is the environment associated with the GitHub Actions run,
+        # if no environment is configured for the GitHub Actions run, this will
+        # be empty.
+        environment: production
+        # actor is the GitHub username that caused the GitHub Actions run,
+        # whether by comitting or by directly despatching the workflow.
+        actor: octocat
+        # ref is the git ref that triggered the action run.
+        ref: ref/heads/main
+        # ref_type is the type of the git ref that triggered the action run,
+        # for example `branch`.
```

Comment:
Is this clause required when the provided example value of the key is the same thing?